### PR TITLE
✨ Add data science certificate creation

### PR DESCRIPTION
**Pull request by phlgr**
The layout and api requests are changed to support the creation of Data Science certificates.
Known issues: Transparent images will get a black background when loaded.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -90,13 +91,19 @@ export type CourseTopics = {
...
 export type Course = {
   id: string;
   coach: string;
+  type: 'web' | 'java' | 'data';
   startDate: string;
   endDate: string;
   location: string;
-  topics: CourseTopics;
+  topics: CourseTopics | DataCourseTopics;
 };
```
Comment (lmachens) [inline src/api.ts:107]: I guess you can not have a course with `type: "web"` and `topics` of type `DataCourseTopics`.
Thus, you could improve the typing here to allow only specific combinations.

As an example:

```ts
export type Course = {
  id: string;
  coach: string;
  startDate: string;
  endDate: string;
  location: string;
} & ({
  type: 'web' | 'java';
  topics: CourseTopics;
} | {
  type: 'data';
  topics: DataCourseTopics;
});
```

Feel free to create sub types like 
```ts
type DataCoursePartial = {
  type: 'data';
  topics: DataCourseTopics;
}
```

File: src/certificate.ts [hunk 2]
```diff
@@ -18,17 +18,18 @@ export async function responseCertificate(
...
   const doc = new PDFDocument({ size: 'A4' });
+
   doc.pipe(res);
 
   renderFirstPage(doc, talent, course);
 
   doc.addPage();
 
-  renderSecondPage(doc, course.topics);
+  renderSecondPage(doc, course.topics, course.type);
```
Comment (lmachens) [inline src/certificate.ts:28]: It's getting more complex if you add more arguments. I would reduce it to a single object argument or even better, only bypass `doc`. 

```suggestion
  renderSecondPage(doc);
```

or 
```suggestion
  renderSecondPage({ 
    doc, 
    topics: course.topics, 
    type: course.type
  });
```

`topics` and `type` are already part of it.
Same for other functions.

File: src/certificate.ts [hunk 3]
```diff
@@ -207,7 +212,11 @@ function renderFirstPage(
   });
 }
 
-function renderSecondPage(doc: PDFKit.PDFDocument, topics: CourseTopics) {
+function renderSecondPage(
+  doc: PDFKit.PDFDocument,
+  topics: CourseTopics | DataCourseTopics,
```
Comment (lmachens) [inline src/certificate.ts:217]: Maybe add a new type called `Topics`.

```suggestion
  topics: Topics,
```

**Review by lmachens — COMMENTED** [review]
Reviewed 50%. Good job so far

**Review by lmachens — APPROVED** [review]
gj

but it's getting more complex and the render-page functions should be split into multiple functions for better readability.
I recommend spending some time in the code structure after this to make sure this projects stays maintainable.

**Comment by phlgr** [comment]
@lmachens I absolutely agree. Refactoring will be the next big step.

File: src/certificate.ts [hunk 4]
```diff
@@ -109,18 +112,20 @@ function renderFirstPage(
...
-      width: A4SIZE[0],
-      align: 'center',
-    },
-    fontSize: 17,
-    fillColor: SECONDARY_TEXT_COLOR,
-    font: 'src/assets/fonts/OpenSans/OpenSans-Light.ttf',
-  });
+  if (course.type !== 'data') {
+    text(doc, {
+      text: '(Web Development)',
```
Comment (badabam) [inline src/certificate.ts:117]: Is it the same text for Java?

**Review by badabam — APPROVED** [review]
Very nice!

I wonder, if it would improve readability to invert the conditions: there is a lot of `if ( type !== data)` which could be `if (type === 'web' || type === 'java')`. For data you also could do `type === data` instead of an `else` which _might_ be easier to follow. But maybe not, I guess it's a thing of preference.

File: src/certificate.ts (hunk 4 above)
Comment (phlgr) [reply src/certificate.ts:117]: No, this is mentioned in issue [#23](https://github.com/neuefische/certificates/issues/23) and will not be adressed in this PR

**Comment by phlgr** [comment]
@badabam Thank you for the review. I clarified the if statements as you suggested.

**Pull request merged**